### PR TITLE
Use mailgun/mailgun-go library

### PR DESCRIPTION
**Pull request by darkowlzz**
I wanted to use this provider to setup mailgun routes, but found out that routes are not supported and with the existing mailgun library (https://github.com/pearkes/mailgun), that can't be implemented. Mailgun has official library for go (https://github.com/mailgun/mailgun-go) which supports routes and all the new features.

This PR replaces the existing underlying mailgun library with mailgun/mailgun-go, keeping the same functionality. This would be followed by more PRs implementing routes and other features.

```
$ make testacc TESTARGS='-run=TestAccMailgunDomain_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccMailgunDomain_Basic -timeout 120m
?   	github.com/terraform-providers/terraform-provider-mailgun	[no test files]
=== RUN   TestAccMailgunDomain_Basic
--- PASS: TestAccMailgunDomain_Basic (14.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-mailgun/mailgun	14.234s
```

Also, I've replaced the vendoring with [`dep`](http://github.com/golang/dep/). Since this is a small repo, I thought it would be good for trying out dep.

### CODE REVIEW

**Comment by darkowlzz** [comment]
Also, upgraded the go version in travis to 1.9.1 because the new vendored packages are compatible with 1.9.1 and were causing the tests to fail.
```
==> Checking that code complies with gofmt requirements...
go test -i $(go list ./... |grep -v 'vendor') || exit 1
# github.com/terraform-providers/terraform-provider-mailgun/vendor/github.com/hashicorp/terraform/config
vendor/github.com/hashicorp/terraform/config/testing.go:9: t.Helper undefined (type *testing.T has no field or method Helper)
GNUmakefile:10: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by darkowlzz** [comment]
ugh! just saw that 2 PRs with the same intent were submitted today #4 #5 coincidence! 😅

File: mailgun/resource_mailgun_domain.go [hunk 1]
```diff
@@ -48,84 +49,28 @@ func resourceMailgunDomain() *schema.Resource {
...
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-						"value": &schema.Schema{
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-					},
-				},
-			},
```
Comment (darkowlzz) [inline mailgun/resource_mailgun_domain.go:100]: Removed these attributes because they cannot be set or updated by any of the APIs.

**Comment by sebglon** [comment]
@radeksimko is it possible to have status and estimate date for a release?
Because this lock new features.

**Comment by wgebis** [comment]
Done in #18

**Pull request closed without merging**